server/bin/pbench-audit-server.sh

```diff
@@ -199,55 +199,77 @@ function verify_incoming {
         > ${lclreport}
 
         tarball_dirs="${workdir}/tarballdirs"
+        > ${tarball_dirs}
         empty_tarball_dirs="${workdir}/emptytarballdirs"
-        > ${empty_tarball_dirs}.unsorted
+        > ${empty_tarball_dirs}
+        unpacking_tarball_dirs="${workdir}/unpackingtarballdirs"
+        > ${unpacking_tarball_dirs}
         tarball_links="${workdir}/tarballlinks"
+        > ${tarball_links}
         find $INCOMING/${controller} -maxdepth 1 \
-                \( -type d ! -name ${controller} ! -empty -fprintf ${tarball_dirs} "%f\n" \) \
-                -o \( -type d ! -name ${controller} -empty -fprintf ${empty_tarball_dirs}.unsorted "\t\t%f\n" \) \
-                -o \( -type l -fprintf ${tarball_links} "%f %l\n" \)
+                   \( -type d ! -name ${controller} ! -name '*.unpack' ! -empty -fprintf ${tarball_dirs} "%f\n" \) \
+                -o \( -type d ! -name ${controller} ! -name '*.unpack'   -empty -fprintf ${empty_tarball_dirs} "\t\t%f\n" \) \
+                -o \( -type d ! -name ${controller}   -name '*.unpack'          -fprintf ${unpacking_tarball_dirs} "\t\t%f\n" \) \
+                -o \( -type l                                                   -fprintf ${tarball_links} "%f %l\n" \)
         status=$?
         if [[ $status -ne 0 ]]; then
             printf "*** ERROR *** unable to traverse $INCOMING/${controller}: find failed with $status" >> ${lclreport}
             let cnt=cnt+1
         fi
 
         invalid_tb_dirs="${workdir}/invalidtbdirs"
-        > ${invalid_tb_dirs}.unsorted
+        > ${invalid_tb_dirs}
         while read tb ; do
             if [[ -r $ARCHIVE/${controller}/${tb}.tar.xz ]]; then
                 continue
             fi
-            printf "\t\t${tb}\n" >> ${invalid_tb_dirs}.unsorted
+            printf "\t\t${tb}\n" >> ${invalid_tb_dirs}
```

Comment:
It'd be more efficient to redirect the while loop to the file, rather than doing it every time through the loop.

Reply:
Done.